File: leetcode/go/3.longest-substring-without-repeating-characters.go
/*
 * @lc app=leetcode id=3 lang=golang
 *
 * [3] Longest Substring Without Repeating Characters
 */

// @lc code=start
func lengthOfLongestSubstring(s string) int {
	if len(s) == 0 {
		return 0
	}
	var find = false
	pos := 0
	loop := 1
	length := 1
	max_length := 1
	for loop < len(s) {
		i := 0
		find = false
		for i < length {
			if s[pos+i] == s[loop] {
				find = true
				length = loop - pos - i
				pos = pos + i + 1
				loop++
				break
			}
			i++
		}
		if find {
			continue
		}
		length++
		loop++
		if length > max_length {
			max_length = length
		}
	}
	return max_length
}

// @lc code=end

File: leetcode/cpp/3.longest-substring-without-repeating-characters.cpp
/*
 * @lc app=leetcode id=3 lang=cpp
 *
 * [3] Longest Substring Without Repeating Characters
 */
// @lc code=start
class Solution
{
public:
    int lengthOfLongestSubstring(string s)
    {
        if (s.empty())
            return 0;
        int loop = 1;
        int pos = 0;
        int max_len = 1;
        int len = 1;
        int i = 0;
        bool find = false;
        while (loop < s.size())
        {
            i = 0;
            find = false;
            while (i < len)
            {
                if (s[loop] == s[pos + i])
                {
                    pos = pos + i + 1;
                    len = loop - pos + 1;
                    ++loop;
                    find = true;
                    break;
                }
                ++i;
            }
            if(find == true){
                continue;
            }
            ++len;
            ++loop;
            if (len > max_len){ 
                max_len = len;
            }
                
        }
        return max_len;
    }
};
// @lc code=end
